### ISSUE TITLE
MapControls doesn't work as expected

### ISSUE THREAD

**Issue by GuskiS** [pmndrs/three-stdlib#47]
- `three` version: 0.126.1
- `@react-three/fiber` version: 6.0.0
- `node` version: 12.18.3
- `npm` (or `yarn`) version: 1.22.5
- browser version: `chrome@latest`/`edge@latest`

### Problem description:
When visiting https://drei.pmnd.rs/?path=/story/controls-mapcontrols--map-controls-scene-st everything works as expected.

When I tried running drei's storybook locally, I get this:
![MapControls](https://user-images.githubusercontent.com/3505777/113345157-bc366600-933a-11eb-9b29-ba21f0765381.gif)

CC: https://github.com/pmndrs/drei/issues/351

**Comment by hasparus** [comment]
I have the same problem — after updating from `react-three-fiber` to `@react-three/fiber` orbit controls don't even respect `enabled` prop.

**Comment by joshuaellis** [comment]
Either of you are welcome to look and make a PR, will be fixed quicker that way.

**Comment by hasparus** [comment]
I'm looking for the cause right now. TBH I'm not sure if the problem originates from here.
I compared OrbitControls in three-stdlib and three/examples and didn't found anything suspicious.

- Downgrading @react-three/drei and aliasing @react-three/fiber to react-three-fiber didn't help.
- Downgrading three to 0.126 helped. :question: :thinking:

**Comment by hasparus** [comment]
Ha! Found the problem. It might be a problem in drei?

I have two copies of MapControls. One accepts props given to MapControls from drei, the other one is created in drei's MapControls useState with default settings.

I commented out `<primitive` call, and I'm back to one.

![image](https://user-images.githubusercontent.com/15332326/113435971-23f4bb80-93e4-11eb-977b-ec9c0e42c66b.png)

This is connected to https://github.com/mrdoob/three.js/issues/20575 and https://github.com/pmndrs/drei/issues/169 as MapControls calls `connect` and registers event handlers, and we end up with side effects in `useState` (Does React guarantee it's called only once? I don't think so.)
